File: Python/2019/d2p2.py
import math

input  = open("input.txt", "r")

input = input.read()

inputList = input.split(',')

for i in range(len(inputList)):
    inputList[i] = int(inputList[i])

def run(arr1, noun, verb):
    arr = []
    for i in range(len(arr1)):
       arr.append(arr1[i])
    arr[1] = noun
    arr[2] = verb
    i = 0
    while(i < len(arr)):
        if(arr[i] == 1):
            arr[arr[i+3]] = arr[arr[i+1]] + arr[arr[i+2]]
        elif(arr[i] == 2):
            arr[arr[i+3]] = arr[arr[i+1]] * arr[arr[i+2]]
        elif(arr[i] == 99):
            return arr[0]
        i = i + 4

for i in range(100):
    for j in range(100):
        if(run(inputList, i, j) == 19690720):
            print(i * 100 + j)
    

File: Javascript/2019/d2p2.js
var fs = require('fs'),
    path = require('path'),    
    filePath = path.join(__dirname, 'input');

fs.readFile(filePath, {encoding: 'utf-8'}, function(err,data){
    if (!err) {
        let arr1 = data.split(',')
        for(var i = 0; i < 100; i++){
            for(var j = 0; j < 100; j++){
                
                arr1[1] = i;
                arr1[2] = j;
                
                if(solvePuzzle(arr1) == 19690720){
                    console.log((i * 100) + j)
                }

                
            }
        }
        
    } else {
        console.log(err);
    }
});

solvePuzzle = (arr1) => {
    let arr = [];
    for(var i = 0; i< arr1.length; i++){
        arr.push(arr1[i])
    }
    
    for(let i = 0; i < arr.length; i += 4){
        if(arr[i] == 1){
            let num = parseInt(arr[arr[i+1]]) + parseInt(arr[arr[i+2]]);
            arr[parseInt(arr[i+3])] = num
        }else if(arr[i] == 2){
            let num = parseInt(arr[arr[i+1]]) * parseInt(arr[arr[i+2]]);
            arr[parseInt(arr[i+3])] = num
        }else if(arr[i] = 99){
            
            return arr[0];
        }
    }
    
}

